pkg/reconciler/broker/broker.go

```diff
@@ -193,7 +173,31 @@ func (r *Reconciler) reconcileBroker(ctx context.Context, b *brokerv1beta1.Broke
 	}
 	b.Status.PropagateIngressAvailability(ingressEndpoints)
 
+	// Filter by `eventing.knative.dev/broker: <name>` here
+	// to get only the triggers for this broker. The trigger webhook will
+	// ensure that triggers are always labeled with their broker name.
+	triggers, err := r.triggerLister.Triggers(b.Namespace).List(labels.SelectorFromSet(map[string]string{eventing.BrokerLabelKey: b.Name}))
+	if err != nil {
+		logger.Error("Problem listing triggers", zap.Error(err))
+		return err
+	}
+
+	r.reconcileConfig(ctx, b, triggers)
+	// Update config map
+	r.flagTargetsForUpdate()
+	return nil
+}
+
+// reconcileConfig reconstructs the data entry for the given broker in targets-config.
+func (r *Reconciler) reconcileConfig(ctx context.Context, b *brokerv1beta1.Broker, triggers []*brokerv1beta1.Trigger) {
+	// TODO Maybe get rid of BrokerMutation and add Delete() and Upsert(broker) methods to TargetsConfig. Now we always
+	//  delete or update the entire broker entry and we don't need partial updates per trigger.
+	// The code can be simplified to r.targetsConfig.Upsert(brokerConfigEntry)
 	r.targetsConfig.MutateBroker(b.Namespace, b.Name, func(m config.BrokerMutation) {
+		// First delete the broker entry.
+		m.Delete()
```

Comment:
See my other comment. Let me fix the `delete` func. Either wait for my change or go ahead and having delete in the separate mutation in the PR (and I will fix that later).

Reply:
https://github.com/google/knative-gcp/pull/1142